tests/test_coreg.py

```diff
@@ -218,64 +218,75 @@ def test_coreg_example(self, verbose: bool = False) -> None:
+    def test_coreg_example_shift(self, shift_px, coreg_class, points_or_raster, verbose=False, downsampling=5000):
         """
         For comparison of coreg algorithms:
-        Shift a ref_dem on purpose, e.g. shift_px = (1,1), and then apply coreg to shift it back.
+        Shift a ref_dem on purpose, e.g. shift_px = (1,1), and then applying coreg to shift it back.
         """
         warnings.simplefilter("error")
-
         res = self.ref.res[0]
 
-        # Shift DEM by shift_px
+        # shift DEM by shift_px
         shifted_ref = self.ref.copy()
         shifted_ref.shift(shift_px[0] * res, shift_px[1] * res)
 
-        for cor in coregs:
-            # Do coreg on shifted DEM
-            if cor == "NuthKaab":
-                print("\n(1) NuthKaab")
-                nuth_kaab = xdem.coreg.NuthKaab()
-                nuth_kaab.fit(shifted_ref, self.ref, inlier_mask=self.inlier_mask, verbose=verbose)
-                assert nuth_kaab._meta["offset_east_px"] == pytest.approx(-shift_px[0], rel=1e-2)
-                assert nuth_kaab._meta["offset_north_px"] == pytest.approx(-shift_px[1], rel=1e-2)
-
-            if cor == "GradientDescending":
-                print("\n(2) GradientDescending")
-                gds = xdem.coreg.GradientDescending(downsampling=downsampling)
-                gds.fit_pts(shifted_ref, self.ref, inlier_mask=self.inlier_mask, verbose=verbose)
-                assert gds._meta["offset_east_px"] == pytest.approx(-shift_px[0], rel=1e-2)
-                assert gds._meta["offset_north_px"] == pytest.approx(-shift_px[1], rel=1e-2)
-
-            if cor == "NuthKaab_pts":
-                print("\n(3) NuthKaab running on pts_fit")
-                nuth_kaab = xdem.coreg.NuthKaab()
-                nuth_kaab.fit_pts(shifted_ref, self.ref, inlier_mask=self.inlier_mask, verbose=verbose)
-                assert nuth_kaab._meta["offset_east_px"] == pytest.approx(-shift_px[0], rel=1e-2)
-                assert nuth_kaab._meta["offset_north_px"] == pytest.approx(-shift_px[1], rel=1e-2)
+        shifted_ref_points = shifted_ref.to_points(as_array=False, subset=downsampling, pixel_offset="center").ds
+        shifted_ref_points["E"] = shifted_ref_points.geometry.x
+        shifted_ref_points["N"] = shifted_ref_points.geometry.y
+        shifted_ref_points.rename(columns={"b1": "z"}, inplace=True)
+
+        kwargs = {} if coreg_class.__name__ != "GradientDescending" else {"downsampling": downsampling}
+
+        coreg_obj = coreg_class(**kwargs)
+
+        best_east_diff = 1e5
+        best_north_diff = 1e5
+        if points_or_raster == "raster":
+            coreg_obj.fit(shifted_ref, self.ref, verbose=verbose)
+        elif points_or_raster == "points":
+            coreg_obj.fit_pts(shifted_ref_points, self.ref, verbose=verbose)
+
+        if coreg_class.__name__ == "ICP":
+            matrix = coreg_obj.to_matrix()
+            coreg_obj._meta["offset_east_px"] = -matrix[0][3] / res
```

Comment:
I'm not sure I understand why you overwrite the `_meta` here for ICP? A comment to explain the logic would be helpful

Reply:
Added comments in [df66946](https://github.com/GlacioHack/xdem/pull/407/commits/df66946a6f614d75e725c7a8bb7430bede0a6142). 

The metadata is not overridden as `offset_east_px` and  `offset_north_px`  don't usually exist.